Formula/iqtree3.rb

```diff
@@ -0,0 +1,83 @@
+class Iqtree3 < Formula
+  # cite Wong_2025: "https://doi.org/10.32942/X2P62N"
+  desc "Efficient and versatile phylogenomic software by maximum likelihood"
+  homepage "http://www.iqtree.org/"
+  # Use git clone to get submodules
+  url "https://github.com/iqtree/iqtree3.git", branch: "master", shallow: false
+  version "3.0.1"
+  license "GPL-2.0-only"
+  head "https://github.com/iqtree/iqtree3.git", branch: "master", shallow: false
+
+  depends_on "boost" => :build
+  depends_on "cmake" => :build
+  depends_on "eigen" => :build
+  depends_on "googletest" if OS.mac? # For test helpers
+  depends_on "libomp" if OS.mac?
+  depends_on "llvm" if OS.mac?
+  depends_on "zlib"
+  depends_on "lsd2" => :optional
```

Comment:
```suggestion
  depends_on "boost" => :build
  depends_on "cmake" => :build
  depends_on "eigen" => :build
  depends_on "brewsci/bio/lsd2"

  uses_from_macos "zlib"

  on_macos do
    depends_on "googletest"
    depends_on "libomp"
    depends_on "llvm"
  end
```

- If there are multiple `if OS.mac?` blocks, please use `on_macos` block instead.
- For most cases, `uses_from_macos "zlib"` is recommended.
- Use `brewsci/bio/lsd2` for brewsci's packages. `=> :optional` is currently discouraged.